core/modules/path/path.path.inc

```diff
@@ -47,6 +48,26 @@ function node_path_info() {
 }
 
 /**
+ * Callback to delete aliases for node types.
+ */
+function node_path_type_delete_callback($types) {
+  $query = db_select('node', 'n');
+  $query->addField('n', 'nid');
+  $query->condition('n.type', $types, 'IN');
+  $nids = $query->execute()->fetchCol();
+
+  $sources = array();
+  foreach ($nids as $nid) {
+    $sources[] = 'node/' . $nid;
+  }
+
+  $query = db_delete('url_alias');
+  $query->condition('source', $sources, 'IN');
```

Comment:
This could cause a problem with anything more than a few thousand items. I was going to recommend a subquery here, but it looks like we need to tack on `node/` onto each path before we do the `IN` condition. So yes... looks like a batch is necessary.